unit/java_bytecode/java_types/java_type_from_string.cpp

```diff
@@ -0,0 +1,48 @@
+/*******************************************************************\
+
+ Module: Unit tests for java_types
+
+ Author: Diffblue Ltd.
+
+\*******************************************************************/
+
+#include <testing-utils/catch.hpp>
+#include <java_types.h>
+
+SCENARIO("java_type_from_string", "[core][java_types]")
+{
+  WHEN("Ljava/lang/Integer;")
+  {
+    const auto generic_symbol_type =
```

Comment:
`generic_symbol_type` isn't a great variable name. If you could also put a TODO in these tests to indicate they aren't very comprehensive :stuck_out_tongue: